microsoft/MixedRealityToolkit-Unity
# Add Active property and Touch state to Interactive Element

## Overview

Added the `Active` property and support for the Touch state to Interactive Element.  

Also added 2 new tests.

#### Touch State
When the Touch state is added via inspector, a Near Interaction Touchable Volume is also added as a Near Interaction Touchable is required for touch input.  If a user adds the Touch state during runtime, a Near Interaction Touchable is added as well. 

Once a button class is added, a Near Interaction Touchable will be attached instead of a Near Interaction Touchable Volume.

![AddTouchState](https://user-images.githubusercontent.com/53493796/99316499-54fb9780-2819-11eb-995f-1e5ac87397e5.gif)

#### Active Property
The `Active` property controls whether or not the Interactive Element updates state changes internally.  The `Active` property maps to Interactable's `Enabled` property.  The naming was changed from `Enabled` to `Active` to limit the confusion between a Monobehaviour's `Enabled` property and Interactable's internal `Enabled` property.

### Interactive Element State Values Runtime Inspector View

Watch the state values change in the inspector for the Focus and Touch state as the input simulation hand moves.
![Focus+TouchState](https://user-images.githubusercontent.com/53493796/99317953-b58bd400-281b-11eb-99d4-01f711da9338.gif)

## Changes
- Completes the add Active property item on this checklist #8883 

## Verification
How to test:
1. Checkout the branch
2. Open new unity scene
3. Add MRTK to the scene
4. Add a cube to the scene
5. Attach the `Interactive Element` component
6. Select Add Core State button in the Interactive Element inspector
7. Select Touch 
8. Press play and observe the state value changes in the inspector
